lib/logger/index.ts

```diff
@@ -0,0 +1,76 @@
+import * as Logger from 'bunyan';
+
+const is = require('@sindresorhus/is');
+const PrettyStdout = require('./pretty-stdout').RenovateStream;
+const configSerializer = require('./config-serializer');
+const errSerializer = require('./err-serializer');
+const cmdSerializer = require('./cmd-serializer');
+
+let meta = {};
+
+const stdout: Logger.Stream = {
+  name: 'stdout',
+  level: (process.env.LOG_LEVEL as Logger.LogLevel) || 'info',
+  stream: process.stdout,
+  type: 'raw',
+};
+
+if (process.env.LOG_FORMAT !== 'json') {
+  const prettyStdOut = new PrettyStdout();
+  prettyStdOut.pipe(process.stdout);
+  stdout.stream = prettyStdOut;
+}
+
+const bunyanLogger: any = Logger.createLogger({
+  name: 'renovate',
+  serializers: {
+    body: configSerializer,
+    cmd: cmdSerializer,
+    config: configSerializer,
+    err: errSerializer,
+  },
+  streams: [stdout],
+});
+
+const logFactory = (level: string): any => {
+  return (p1: any, p2: any): Logger => {
+    (global as any).renovateError =
+      (global as any).renovateError || level === 'error' || level === 'fatal';
```

Comment:
@rarkins Maybe we should store all `error` and `fatal` to be logged at renovate end too. Then we can find the cause more easy, because we do not have to search the whole log.

Reply:
That could be useful. Then logic at the end of global worker could then be something like this:

```
const loggerErrors = logger.getErrors();
if (loggerErrors.length) {
	logger.info({ loggerErrors }, "Renovate is exiting with a non-zero code due to the following logged errors");
}

```

BTW we don't necessarily need to store these errors on `global` either - the logger module could store them and provide a getter function